code/game/machinery/buttons.dm

```diff
@@ -10,6 +10,52 @@
 	idle_power_usage = 2
 	active_power_usage = 4
 
+/obj/machinery/driver_button/secure
+	name = "secure mass driver button"
+	icon_state = "is_closed"
+	desc = "A secure remote control switch for a mass driver."
+	var/closed = 1
+
+/obj/machinery/driver_button/secure/north

```

Comment:
Don't make mapping instances like this. Mappers can generate instances from icon states or directions, and set their pixel_x/pixel_y by hand.